tests/rptest/tests/datalake/datalake_dlq_test.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from typing import Optional
+
+from ducktape.mark import matrix
+
+from rptest.clients.serde_client_utils import SchemaType, SerdeClientType
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import (
+    PandaproxyConfig,
+    SchemaRegistryConfig,
+    SISettings,
+)
+from rptest.services.serde_client import SerdeClient
+from rptest.tests.datalake.datalake_services import DatalakeServices
+from rptest.tests.datalake.query_engine_base import QueryEngineBase, QueryEngineType
+from rptest.tests.datalake.utils import supported_storage_types
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class DatalakeDLQTest(RedpandaTest):
+    def __init__(self, test_ctx, *args, **kwargs):
+        super(DatalakeDLQTest,
+              self).__init__(test_ctx,
+                             num_brokers=1,
+                             si_settings=SISettings(test_context=test_ctx),
+                             extra_rp_conf={
+                                 "iceberg_enabled": "true",
+                                 "iceberg_catalog_commit_interval_ms": 5000
+                             },
+                             schema_registry_config=SchemaRegistryConfig(),
+                             pandaproxy_config=PandaproxyConfig(),
+                             *args,
+                             **kwargs)
+        self.test_ctx = test_ctx
+        self.topic_name = "test"
+
+    def setUp(self):
+        # redpanda will be started by DatalakeServices
+        pass
+
+    def _get_serde_client(
```

Comment:
nit: in case you find it easier to use, I've recently preferred using confluent_kakfa so as a test author I have more control of the input schema. No need to change if you prefer this

Reply:
This is copy & pasted (meh) from the e2e test. I will keep this in mind and try to extract this into something nicer when I'll have all DLQ functionality in.